modules/apps/portal-search/portal-search-rest-test/src/testIntegration/java/com/liferay/portal/search/rest/resource/v1_0/test/SearchResultResourceTest.java

```diff
@@ -1131,6 +1145,91 @@ private void _testPostSearchPageWithKeywords() throws Exception {
 		Assert.assertEquals(1L, searchPage.getTotalCount());
 	}
 
+	private void _testPostSearchPageWithLocalizedTextFields() throws Exception {
+		PrincipalThreadLocal.setName(_user.getUserId());
+		PermissionThreadLocal.setPermissionChecker(
+			PermissionCheckerFactoryUtil.create(_user));
+
+		String objectFieldName = "localized" + RandomTestUtil.randomString();
+
+		_objectDefinition =
+			_objectDefinitionLocalService.addCustomObjectDefinition(
+				_user.getUserId(), 0, null, false, true, true, false,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				ObjectDefinitionTestUtil.getRandomName(), null, null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				true, ObjectDefinitionConstants.SCOPE_COMPANY,
+				ObjectDefinitionConstants.STORAGE_TYPE_DEFAULT, null);
+
+		ObjectFieldUtil.addCustomObjectField(
+			new TextObjectFieldBuilder(
+			).userId(
+				_user.getUserId()
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).indexed(
+				true
+			).name(
+				objectFieldName
+			).objectDefinitionId(
+				_objectDefinition.getObjectDefinitionId()
+			).localized(
+				true
+			).build());
+
+		ObjectFieldUtil.addCustomObjectField(
+			new TextObjectFieldBuilder(
+			).userId(
+				_user.getUserId()
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).indexed(
+				true
+			).name(
+				"foo"
+			).objectDefinitionId(
+				_objectDefinition.getObjectDefinitionId()
+			).build());
```

Comment:
This way is enough, we only have to test a localized field and we can pass a list in the object definition creation, please rerun this test again to ensure it works well

```suggestion
		_objectDefinition =
			_objectDefinitionLocalService.addCustomObjectDefinition(
				_user.getUserId(), 0, null, false, true, true, false,
				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
				ObjectDefinitionTestUtil.getRandomName(), null, null,
				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
				true, ObjectDefinitionConstants.SCOPE_COMPANY,
				ObjectDefinitionConstants.STORAGE_TYPE_DEFAULT,
				Arrays.asList(
					new TextObjectFieldBuilder(
					).labelMap(
						LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
					).indexed(
						true
					).name(
						"localizedTextObjectFieldName"
					).localized(
						true
					).build()
				));
```